evalml/model_understanding/_partial_dependence_fast_mode_utils.py

```diff
@@ -0,0 +1,105 @@
+    if is_regression(pipeline.problem_type):
+        # --> might want to use random seed here, though the mocked y shouldn't have an impact on PD
+        mock_y = pd.Series(np.random.randint(0, 10, len_X))
+    elif is_binary(pipeline.problem_type):
+        mock_y = pd.Series(np.random.choice([True, False], size=len_X))
+    else:
+        mock_y = pd.Series(np.random.choice([0, 1, 2], size=len_X))
+
+    # Create a fit pipeline for each feature
+    cloned_feature_pipelines = {}
+    for variable in features:
+        # Don't fit pipelines if the feature has no impact on predictions
+        if not variable_has_features_passed_to_estimator[variable]:
+            continue
+        pipeline_copy = pipeline.new(
+            parameters=new_parameters,
+        )
+        pipeline_copy.fit(X.ww[[variable]], mock_y)
+        cloned_feature_pipelines[variable] = pipeline_copy
+
+    return cloned_feature_pipelines
+
+
+def _transform_single_feature(
+    X,
+    X_t,
+    feature_provenance,
+    variable,
+    part_dep_column,
+    cloned_pipeline,
+):
+    """Transforms single column using cloned pipeline and column that has been updated
+        for partial dependence calculations.
+
+    Args:
+        X_t (pd.DataFrame, np.ndarray): The transformed data into which we insert the transformed single column.
+        X (pd.DataFrame, np.ndarray): The original data used for calculating partial dependence.
+        feature_provenance (dict[str, set[str]]): Dictionary mapping base feature names to engineered feature names.
+        variable (str): The name of the single column we're transforming.
+        part_dep_column (pd.Series): The updated column that will be transformed by the fit pipeline.
+        cloned_pipeline (PipelineBase or subclass): Fitted pipeline that will be used to transform a single feature.
+
+    Returns:
+        dict[str, PipelineBase or subclass]: Dictionary mapping feature name to the pipeline pipeline
+            fit for it.
+    """
+    changed_col_df = pd.DataFrame({variable: part_dep_column})
+    changed_col_df.ww.init(
+        logical_types={variable: X.ww.logical_types[variable]},
+    )
+
+    # Take the changed column and send it through transform by itself
+    X_t_single_col = cloned_pipeline.transform_all_but_final(changed_col_df)
+
+    # Determine which columns in X_t we're replacing with the transform output
+    cols_to_replace = [variable]
+    if feature_provenance.get(variable):
+        # cols to replace has to be in the same order as X_t
+        cols_to_replace = [
+            col for col in X_t if col in feature_provenance.get(variable)
+        ]
+    # If some categories got dropped during transform of the original X_t,
+    # they won't be in X_t_single_col, so don't include them
+    if len(cols_to_replace) != len(X_t_single_col.columns):
+        X_t_single_col = X_t_single_col[cols_to_replace]
+
+    # --> not keeping in woodwork - problematic?
+    X_t[cols_to_replace] = X_t_single_col
```

Comment:
I wasn't sure if I should be adding the columns into X_t though woodwork or not. This X_t gets passed to `estimator.predict`, so I would think that it'd be problematic if the woodwork schema was invalid. What I _think_ is happening here to allow this to work is that the `cols_to_replace` are the correct types to not break the woodwork schema. 

But maybe I should just go through woodwork to be safe. Open to hearing other peoples' thoughts!